src/main/people.ts

```diff
@@ -0,0 +1,58 @@
+import { SD_TEAM } from "../../config/sdTeam";
+import { getBasecampUrl, sendPaginatedBasecampGetRequest } from "./basecamp";
+import { getScriptProperty, setScriptProperties } from "./propertiesService";
+
+type PersonNameIdMap = {[key: string]: string};
+
+const PEOPLE_PATH: string = "/people.json";
+const PEOPLE_POPULATED_KEY = "peoplePopulated";
+
+/**
+ * Utility function that retrieves people from the Basecamp API, selects SD people, and then populates
+ * the script properties with a map from person name to Basecamp person id. Can be invoked manually
+ * from the Google Apps Script Developer UI
+ */
+export function populatePeopleInDb(): void {
+    const requestUrl = getBasecampUrl() + PEOPLE_PATH;
+    const peopleData: Person[] = sendPaginatedBasecampGetRequest(requestUrl) as Person[];
+    
+    // Filter people by SD; currently this is a no-op
+    const sdPeople: Person[] = peopleData.filter((person) => SD_TEAM.has(person.name));
+
+    // Reduce all of the people to a single map
+    const personNameIdMap: PersonNameIdMap = sdPeople.reduce((map, person) => {
+        map[person.name] = person.id;
+        return map;
+    }, {} as PersonNameIdMap);
+
+    personNameIdMap[PEOPLE_POPULATED_KEY] = "true";
+
+    setScriptProperties(personNameIdMap);
```

Comment:
i just realized, one downside of saving a map of personName -> personId is that when ppl are removed from the sd list, we'd have to manually call a function to clean up that specific person from properties service otherwise it'll remain there forever. similar issue with removing rowIds too i think.
instead, what do you think of storing the whole people map/object as the value, and the key would be a constant like "PEOPLE_MAP"? then we could more easily do a full replacement of the people map object and not worry about losing any ppl references. the downside here is if we were to retrieve the whole people map object each time it'd be slow, but we could cache it in memory by retrieving the full object once. what are your thoughts?

Reply:
Script properties actually have a UI that allows you to delete individual values, but I am ok storing the whole map/object and using an in memory cache. Makes the script properties a lot cleaner especially if we are going to store under values here too. I don't think the in memory cache would persist between different triggers but within one trigger we would only need to fetch the map once

Reply:
Should we also consider doing something similar for the row mapping object as well?